remove deprecation warning

ggreif opened the pull request:
Eliminates following warnings:
```
      ocamlc src/frontend/.ocamlmerlin_server.eobjs/old_protocol.{cmi,cmo,cmt}
File "src/frontend/old/old_protocol.ml", line 97, characters 16-25:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
      ocamlc src/frontend/.ocamlmerlin_server.eobjs/old_IO.{cmi,cmti}
File "src/frontend/old/old_IO.mli", line 37, characters 22-31:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
File "src/frontend/old/old_IO.mli", line 39, characters 48-57:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
File "src/frontend/old/old_IO.mli", line 44, characters 33-42:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
File "src/frontend/old/old_IO.mli", line 44, characters 54-63:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
File "src/frontend/old/old_IO.mli", line 49, characters 33-42:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
File "src/frontend/old/old_IO.mli", line 49, characters 54-63:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
copying path '/nix/store/7rcw8y4x22dmrm0yjjqyxqc020bmssvv-ocaml4.06.1-atdgen-2.0.0' from 'https://nix.dfinity.systems'...
copying path '/nix/store/kvgsbab4f6sfi39pkbvi385j5s2nd3v2-ocaml4.06.1-ppx_tools_versioned-5.1' from 'https://cache.nixos.org'...
    ocamlopt src/frontend/.ocamlmerlin_server.eobjs/old_protocol.{cmx,o}
File "src/frontend/old/old_protocol.ml", line 97, characters 16-25:
Warning 3: deprecated: Std.Json.json
json types are being renamed and will be removed in the next Yojson major version. Use type t instead
installing
```

trefis commented:
What about: https://github.com/ocaml-community/yojson/blob/master/CHANGES.md#fix ?

ggreif commented:
It is still the way to go forward: https://github.com/ocaml-community/yojson/pull/82#issuecomment-498285661

trefis commented:
Alright.

But if we are to proceed with this PR, then I believe we at least need a lower bound constraint on yojson in the opam file, and to make sure that we're not breaking the lsp version (which uses ppx_yojson_conv IIUC).

ggreif commented:
No haste. I'll figure out all these pieces.

On 9/23/19, Thomas Refis <notifications@github.com> wrote:
> Alright.
>
> But if we are to proceed with this PR, then I believe we at least need a
> lower bound constraint on yojson in the opam file, and to make sure that
> we're not breaking the lsp version (which uses ppx_yojson_conv IIUC).
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/ocaml/merlin/pull/1027#issuecomment-534010585

ggreif commented:
@trefis Added lower bounds, updated download packages and removed a few more deprecation warnings. PTAL!

It should be squashed on merge, or should I do it?

trefis commented:
Thanks!
